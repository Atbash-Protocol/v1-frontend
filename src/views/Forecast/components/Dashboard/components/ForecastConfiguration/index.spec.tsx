import { fireEvent, render, screen } from '@testing-library/react';

import Configuration from '.';

describe('Configuration', () => {
    const props = {
        onConfigurationChange: jest.fn(),
        initialData: {
            stakedSBAmount: '122.01',
            BASHPriceAtPurchase: '80',
            rewardYieldPercent: '120',
            futureBASHMarketPrice: '1200',
        },
    };

    it('renders', () => {
        const comp = render(<Configuration {...props} />);

        expect(comp).toMatchSnapshot();
    });

    it('fires an configurationChange event', async () => {
        const { container } = render(<Configuration {...props} />);

        const input = await container.querySelector('input');
        fireEvent.change(input!, { target: { value: '20' } });

        expect(props.onConfigurationChange).toHaveBeenCalledWith({ ...props.initialData, stakedSBAmount: '20' });
    });
});
