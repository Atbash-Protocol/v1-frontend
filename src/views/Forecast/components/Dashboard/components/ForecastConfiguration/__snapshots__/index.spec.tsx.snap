// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-vzmj5c"
      >
        <div
          class="MuiBox-root css-1by3yj5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
          >
            globe:StakedSBAmount
          </p>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              inputmode="numeric"
              name="stakedSBAmount"
              pattern="^[0-9]*[.,]?[0-9]*$"
              placeholder="globe:EnterStakedBASHAmount"
              type="number"
              value="122.01"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
            >
              <div
                class="MuiBox-root css-og7htl"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Max
                </p>
              </div>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-1yuhvjn"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
          >
            globe:RewardYieldPercent
          </p>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              inputmode="numeric"
              name="rewardYieldPercent"
              pattern="^[0-9]*[.,]?[0-9]*$"
              placeholder="globe:EnterRewardYieldPercent"
              type="number"
              value="120"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
            >
              <div
                class="MuiBox-root css-og7htl"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  globe:Current
                </p>
              </div>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-1yuhvjn"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
          >
            globe:BASHPriceAtPurchase
          </p>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              inputmode="numeric"
              name="BASHPriceAtPurchase"
              pattern="^[0-9]*[.,]?[0-9]*$"
              placeholder="globe:EnterBuyPrice"
              type="number"
              value="80"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
            >
              <div
                class="MuiBox-root css-og7htl"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  globe:Current
                </p>
              </div>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-1yuhvjn"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
          >
            globe:FutureBASHMarketPrice
          </p>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              inputmode="numeric"
              name="futureBASHMarketPrice"
              pattern="^[0-9]*[.,]?[0-9]*$"
              placeholder="globe:EnterFuturePrice"
              type="number"
              value="1200"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
            >
              <div
                class="MuiBox-root css-og7htl"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  globe:Current
                </p>
              </div>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-vzmj5c"
    >
      <div
        class="MuiBox-root css-1by3yj5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
        >
          globe:StakedSBAmount
        </p>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            inputmode="numeric"
            name="stakedSBAmount"
            pattern="^[0-9]*[.,]?[0-9]*$"
            placeholder="globe:EnterStakedBASHAmount"
            type="number"
            value="122.01"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              class="MuiBox-root css-og7htl"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Max
              </p>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
        >
          globe:RewardYieldPercent
        </p>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            inputmode="numeric"
            name="rewardYieldPercent"
            pattern="^[0-9]*[.,]?[0-9]*$"
            placeholder="globe:EnterRewardYieldPercent"
            type="number"
            value="120"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              class="MuiBox-root css-og7htl"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                globe:Current
              </p>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
        >
          globe:BASHPriceAtPurchase
        </p>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            inputmode="numeric"
            name="BASHPriceAtPurchase"
            pattern="^[0-9]*[.,]?[0-9]*$"
            placeholder="globe:EnterBuyPrice"
            type="number"
            value="80"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              class="MuiBox-root css-og7htl"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                globe:Current
              </p>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
        >
          globe:FutureBASHMarketPrice
        </p>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-5z7mkc-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            inputmode="numeric"
            name="futureBASHMarketPrice"
            pattern="^[0-9]*[.,]?[0-9]*$"
            placeholder="globe:EnterFuturePrice"
            type="number"
            value="1200"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              class="MuiBox-root css-og7htl"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                globe:Current
              </p>
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
